core/domain/summary_services_test.py

```diff
@@ -879,3 +879,132 @@ def test_requiring_public_activities_to_be_public_succeeds(self):
+                'new_value': 'Exploration 3 Albert title'
+            }, {
+                'cmd': 'edit_exploration_property',
+                'property_name': 'objective',
+                'new_value': 'An objective 3'
+            }], 'Changed title and objective.')
+
+        self.save_new_valid_exploration(self.EXP_ID4, self.bob_id)
+
+        exp_services.update_exploration(
+            self.bob_id, self.EXP_ID4, [{
+                'cmd': 'edit_exploration_property',
+                'property_name': 'title',
+                'new_value': 'Exploration 4 Bob title'
+            }, {
+                'cmd': 'edit_exploration_property',
+                'property_name': 'objective',
+                'new_value': 'An objective 4'
+            }], 'Changed title and objective.')
+
+        self.save_new_valid_exploration(self.EXP_ID5, self.albert_id)
+
+        exp_services.update_exploration(
+            self.albert_id, self.EXP_ID5, [{
+                'cmd': 'edit_exploration_property',
+                'property_name': 'title',
+                'new_value': 'Exploration 5 Albert title'
+            }, {
+                'cmd': 'edit_exploration_property',
+                'property_name': 'objective',
+                'new_value': 'An objective 5'
+            }], 'Changed title and objective.')
+
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID1)
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID2)
+        rights_manager.publish_exploration(self.albert_id, self.EXP_ID3)
+        rights_manager.publish_exploration(self.bob_id, self.EXP_ID4)
+
+    def test_get_exploration_metadata_dicts(self):
+        summaries = (summary_services.get_exploration_metadata_dicts(
+            [self.EXP_ID1, self.EXP_ID2, self.EXP_ID3], self.albert_id))
+
+        expected_summary1 = {
+            'id': self.EXP_ID1,
+            'objective': u'An objective 1',
+            'title': u'Exploration 1 Albert title',
+        }
+        expected_summary2 = {
+            'id': self.EXP_ID2,
+            'objective': u'An objective 2',
+            'title': u'Exploration 2 Albert title',
+        }
+        expected_summary3 = {
+            'id': self.EXP_ID3,
+            'objective': u'An objective 3',
+            'title': u'Exploration 3 Albert title',
+        }
+        self.assertDictContainsSubset(expected_summary1, summaries[0])
+        self.assertDictContainsSubset(expected_summary2, summaries[1])
+        self.assertDictContainsSubset(expected_summary3, summaries[2])
+
+    def test_user_validity(self):
+        summaries = (summary_services.get_exploration_metadata_dicts(
+            [self.EXP_ID5, self.EXP_ID4], self.bob_id))
+
+        expected_summary = {
+            'id': self.EXP_ID4,
+            'objective': u'An objective 4',
+            'title': u'Exploration 4 Bob title',
+        }
+        self.assertDictContainsSubset(expected_summary, summaries[0])
```

Comment:
How about other cases?
- User A can see public exps of user B in the search results.
- The input data contains exp ids that don't actually correspond to real explorations.
- The input data contains exp ids that have been deleted.